doc/devices/dts/flash_partitions.rst

```diff
@@ -0,0 +1,44 @@
+.. _flash_partitions:
+
+Flash partitions nodes explanation
+##################################
+
+In each of the <boards>.dts files, the flash partitions layout can be described.
+It must be put inside the 'partitions' node in the 'flash0' node.
+
+Mcuboot related partitions
+**************************
+
+boot_partition - This is the partition where the bootloader is expected to be
```

Comment:
This would read better if presented as:

     **boot_partition**
       This is the partition where the bootloader is expected.
       Mcuboot is linked into this partition.

     **slot0_partition**
       This is the partition where the current executable application
       image is expected.  Any bootable application must be linked
       into this area.

     **slot1_partition**
       This is the partition where the downloaded application image
       is collected.  Mcuboot can load this image to slot0_partition and 
       run it.  The slot0_partition and slot1_partition must be the same size. 

     **scratch_partition**
       This partition is used for image swapping support.  See the
       `mcuboot documentation`_ for more details.

(I also made a few suggested edits to the text too.)